FirebaseDatabase/Sources/Utilities/FNextPushId.m

```diff
@@ -59,4 +65,43 @@ + (NSString *)get:(NSTimeInterval)currentTime {
     return [NSString stringWithString:id];
 }
 
++ (NSString *)successor:(NSString *_Nonnull)key {
+    NSInteger keyAsInt;
+    if ([FUtilities tryParseString:key asInt:&keyAsInt]) {
+        if (keyAsInt == INTEGER_32_MAX) {
+            return MIN_PUSH_CHAR;
+        }
+        return [NSString stringWithFormat:@"%ld", (long)keyAsInt + 1];
+    }
+    NSMutableString *next = [NSMutableString stringWithString:key];
+    if ([next length] < MAX_KEY_LEN) {
+        [next insertString:MIN_PUSH_CHAR atIndex:[key length]];
+        return next;
+    }
+
+    long i = [next length] - 1;
+    while (i >= 0) {
+        if ([next characterAtIndex:i] != [MAX_PUSH_CHAR characterAtIndex:0]) {
+            break;
+        }
+        i--;
```

Comment:
```suggestion
        --i;
```

From the Google style guide: "Use the prefix form (++i) of the increment and decrement operators unless you need postfix semantics."